field/TinyLang/Field/Evaluator.hs

```diff
@@ -144,22 +146,37 @@ evalBinOp Mul x y = return . UniConst Field $ x `mul` y
 evalBinOp Div x y = UniConst Field <$> divEval x y
 evalBinOp BAt x y = asIntegerEval x >>= fmap (UniConst Bool) . flip atEval y . fromIntegral
 
-evalStatementUni
-    :: (MonadEvalError f m, Eq f, Field f, AsInteger f)
-    => Env (SomeUniConst f) -> Statement f -> Expr f a -> m (UniConst f a)
-evalStatementUni env (ELet (UniVar _ var) def) rest = do
+evalProgramUni :: (MonadEvalError f m, Eq f, Field f, AsInteger f)
+    => Env (SomeUniConst f) -> Program f -> m (Env (SomeUniConst f))
```

Comment:
It might be useful downstream for `evalProgramUni` to return an entire environment, but we're not using this currently and we can always add it later. I propose to make this function return `m ()` rather than `m (Env (SomeUniConst f))` just so we're not blocked on something that is not currently necessary.

Reply:
Hmm, we are using it in testing at the moment, albeit it does cause some issues as we need to fix our semantics of renamings.